spec/unit/provider/package/powershell_spec.rb

```diff
@@ -0,0 +1,367 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require "chef/mixin/powershell_out"
+
+describe Chef::Provider::Package::Powershell do
+  include Chef::Mixin::PowershellOut
+  let(:timeout) { 900 }
+
+  let(:new_resource) { Chef::Resource::PowershellPackage.new("windows_test_pkg") }
+
+  let(:provider) do
+    node = Chef::Node.new
+    events = Chef::EventDispatch::Dispatcher.new
+    run_context = Chef::RunContext.new(node, {}, events)
+    Chef::Provider::Package::Powershell.new(new_resource, run_context)
+  end
+
+  let(:installed_package_stdout) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxCertificate                   2.1.0.0          PSGallery        This module includes DSC resources that simplify administration of certificates on a Windows Server"
+    EOF
+  end
+
+  let(:package_version_stdout) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxCertificate                   2.1.0.0          https://www.powershellgallery... PowerShellGet"
+    EOF
+  end
+
+  let(:installed_package_stdout_xnetworking) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxNetworking                   2.12.0.0          PSGallery        Module with DSC resources for Networking Area"
+    EOF
+  end
+
+  let(:package_version_stdout_xnetworking) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxNetworking                   2.12.0.0          https://www.powershellgallery... PowerShellGet"
+    EOF
+  end
+
+  describe "#initialize" do
+    it "should return the correct class" do
+      expect(provider).to be_kind_of(Chef::Provider::Package::Powershell)
+    end
+  end
+
+  describe "#candidate_version" do
+
+    it "should set the candidate_version to the latest version when not pinning" do
+      my_double = double("powershellout_double")
+      allow(my_double).to receive(:stdout).and_return(package_version_stdout)
```

Comment:
there's a lot of duplication in the doubles in this file, and the variable names aren't very descriptive. Can you try to remove duplication and make the difference between these more clear?

On idea would be to do something like this to set up common doubles early.
```
let(:package_xnetworking_installed) do 
  stdout =  <<-EOF
"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxNetworking                   2.12.0.0          PSGallery        Module with DSC resources for Networking Area"
EOF

  double("powershell_out", :stdout => stdout)
end

let(:package_xnetworking_not_installed) do
  double("powershell_out", :stdout => "")
end
```